Updates of some repo URLs that changed, some licenses and slight corrections of original spellings

Trilarion opened the pull request:
Corrected some original spellings (see their Wikipedia pages for verification): AquaStax -> Aquastax, Mega Lo Mania -> Mega-Lo-Mania, Outrun -> Out Run and Simcity to SimCity. Also corrected within each game entry where these originals were referenced.

Corrected some licenses, are GPL3 but were marked GPL2 (may have been upgraded in between): see http://svn.code.sf.net/p/lgames/code/trunk/lbreakouthd/COPYING and http://svn.code.sf.net/p/rescue/code/Rescue/gpl.txt

And updated some URLs (repos mostly) that have changed. Simply follow the old URLs to see that they were outdated.

I haven't changed the 'updated' field. Would do that after the PR has been reviewed.

cxong commented on games/0.yaml:
```diff
@@ -71,8 +71,8 @@
   type: remake
-  repo: https://github.com/2006rebotted/2006rebotted
-  url: https://rsrebotted.com/
+  repo: https://github.com/2006-Scape/2006Scape
```
need to rename this game to 2006Scape too

Trilarion replied:
Sure

cxong commented on games/d.yaml:
```diff
@@ -132,7 +132,7 @@
   originals:
   - Quake
-  repo: https://svn.icculus.org/twilight/trunk/darkplaces/
+  repo: https://gitlab.com/xonotic/darkplaces.git
```
```suggestion
  repo: https://gitlab.com/xonotic/darkplaces
```

cxong commented on games/h.yaml:
```diff
@@ -86,7 +86,7 @@
   originals:
   - Story of Seasons series
-  repo: https://github.com/greentwip/HarvestMoon
+  repo: https://github.com/Greentwip/HoneyTown
```
also rename to Honey Town

Trilarion replied:
sure

cxong commented on games/l.yaml:
```diff
@@ -82,7 +82,7 @@
   development: active
```
```suggestion
  development: halted
```
let's update this as well; repo is archived

cxong commented on games/o.yaml:
```diff
@@ -1778,8 +1778,8 @@
   - GPL3
-  url: https://github.com/rage8885/OpenSC2K
-  repo: https://github.com/rage8885/OpenSC2K
+  url: https://github.com/nicholas-ochoa/OpenSC2K
```
```suggestion
```
repo and url are the same, so we don't need both

cxong commented on games/r.yaml:
```diff
@@ -406,7 +406,7 @@
   - Java
   license:
-  - GPL2
+  - GPL3
```
please also change the repo to https://sourceforge.net/projects/rescue/

Trilarion replied:
Sure

Trilarion commented:
Thanks for looking over it.

Trilarion commented:
Updated 'updated' fields (except for very small changes like trailing slash or name change of original) and changed url field of Gift Grabber (gave 404 before). From my side ready to merge.
